napari/utils/events/types.py

```diff
@@ -1,8 +1,19 @@
-from typing import Protocol, runtime_checkable
+from typing import Any, Protocol, runtime_checkable
 
 from .event import EmitterGroup
 
 
 @runtime_checkable
 class SupportsEvents(Protocol):
+    # note that if this gets *any* object with an `events` attribute, it will pass
+    # always because type hints are not checked!
+    # this happened with some dask objects, so this is a weak spot in some cases
     events: EmitterGroup
+
+
+class EventedMutable(SupportsEvents, Protocol):
+    def _update_inplace(self, other: Any) -> None:
+        ...
+
+    def _uneventful(self) -> Any:
```

Comment:
kinda seems like something that could be accomplished with an event blocker, rather than requiring each instance of an evented mutable to implement their own event-free mutation?  Also, a docstring explaining what the protocol expects from each of these methods would be good. (`uneventful` suggests "no events" to me... but beyond that, I'd need to go look at the implementations to determine what it's supposed to actually _do_)

Reply:
ohh... after reading some of the implementations, I see now that this is supposed to return a new object, that mimics the protocol of the current object, but without events?  (i.e. EventedList -> List)?  So, it obviously can't be done with an event blocker... but a docstring should be added.